src/cpu/aarch64/acl_gemm_convolution.cpp

```diff
@@ -14,30 +14,142 @@
-status_t acl_gemm_convolution_fwd_t<src_type, wei_type, dst_type,
-        bia_type>::execute_forward(const exec_ctx_t &ctx) const {
-    // Lock here is needed because resource_mapper does not support
-    // concurrent multithreaded access.
-    std::lock_guard<std::mutex> _lock {this->mtx};
-    // Retrieve primitive resource and configured Compute Library objects
-    auto *acl_resource = ctx.get_resource_mapper()->get<acl_resource_t>(this);
-    acl_obj_t<arm_compute::NEGEMMConvolutionLayer> &acl_obj
-            = acl_resource->get_acl_obj();
-
-    return execute_forward_conv_acl<
-            acl_obj_t<arm_compute::NEGEMMConvolutionLayer>, pd_t, src_data_t,
-            wei_data_t, dst_data_t, bia_data_t>(ctx, acl_obj, pd());
+namespace {
+// Keys are anonymous. So deduce the type automagically.
+using conv_key_t = decltype(memory_tracking::names::key_gemm_tmp_buffer);
+
+// Map: [slot , key]
+const std::map<int, conv_key_t> gemm_conv_keys
+        = {{0, conv_key_t::key_gemm_asm_tmp_buffer},
+                {1, conv_key_t::key_gemm_pretranspose_b},
+                {2, conv_key_t::key_gemm_pretranspose},
+                {3, conv_key_t::key_gemm_interleaved_lhs},
+                {4, conv_key_t::key_gemm_pretransposed_rhs},
+                {5, conv_key_t::key_gemm_transposed_1xwrhs},
+                {6, conv_key_t::key_gemm_tmp_buffer},
+                {7, conv_key_t::key_gemm_mm_result_s32},
+                {8, conv_key_t::key_gemm_mm_signed_a},
+                {9, conv_key_t::key_gemm_mm_signed_output},
+                {10, conv_key_t::key_conv_gemm_col},
+                {11, conv_key_t::key_conv_permuted_weights},
+                {12, conv_key_t::key_gemm_output}};
+} // namespace
+
+template <data_type_t src_t, data_type_t wei_t, data_type_t dst_t,
+        data_type_t bia_t>
+status_t
+acl_gemm_convolution_fwd_t<src_t, wei_t, dst_t, bia_t>::create_resource(
+        engine_t *engine, resource_mapper_t &mapper) const {
+    CHECK(pd()->post_ops.create_resource(engine, mapper));
+    return status::success;
+}
+
+template <data_type_t src_t, data_type_t wei_t, data_type_t dst_t,
+        data_type_t bia_t>
+bool acl_gemm_convolution_fwd_t<src_t, wei_t, dst_t,
+        bia_t>::pd_t::zero_points_ok() const {
+    using namespace data_type;
+    // TODO: add support for asymmetric quantization
+    return attr()->zero_points_.has_default_values();
+}
+
+template <data_type_t src_t, data_type_t wei_t, data_type_t dst_t,
+        data_type_t bia_t>
+bool acl_gemm_convolution_fwd_t<src_t, wei_t, dst_t,
+        bia_t>::pd_t::output_scales_mask_ok() const {
+    using namespace data_type;
+    const auto &mask = attr()->output_scales_.mask_;
+    return IMPLICATION(!utils::one_of(src_t, s8, u8),
+                   attr()->output_scales_.has_default_values())
+            // TODO: add support for per_channel quantization
+            && mask == 0;
+}
+
+template <data_type_t src_t, data_type_t wei_t, data_type_t dst_t,
```

Comment:
Kudos for moving things out of the utils files and into cpp, that always annoyed me 😄